core/chains/evm/bulletprooftxmanager/transmitchecker.go

```diff
@@ -0,0 +1,160 @@
+package bulletprooftxmanager
+
+import (
+	"context"
+	"math/big"
+
+	"github.com/ethereum/go-ethereum/accounts/abi/bind"
+	"github.com/ethereum/go-ethereum/common/hexutil"
+	"github.com/pkg/errors"
+
+	evmclient "github.com/smartcontractkit/chainlink/core/chains/evm/client"
+	"github.com/smartcontractkit/chainlink/core/internal/gethwrappers/generated/vrf_coordinator_v2"
+	"github.com/smartcontractkit/chainlink/core/logger"
+	"github.com/smartcontractkit/chainlink/core/utils"
+)
+
+var (
+	_ TransmitCheckerFactory = &CheckerFactory{}
+	_ TransmitChecker        = NoChecker{}
+	_ TransmitChecker        = &SimulateChecker{}
+	_ TransmitChecker        = VRFV2Checker{}
```

Comment:
Minor nit, but why are some of these pointers and some not? Would be nice if they're consistently pointer receivers / value receivers.

Reply:
Yeah true, fixed. VRFV2Checker should be a pointer since it has a member var and we don't want the struct copied whenever it's passed in. 

Switched NoChecker to a singleton here since it's an empty struct and everything can just share the same (non-pointer) version so I think now things should be more clear here.